ksql-version-metrics-client/src/main/java/io/confluent/ksql/version/metrics/KsqlVersionCheckerAgent.java

```diff
@@ -59,7 +61,8 @@ public void start(final KsqlModuleType moduleType, final Properties ksqlProperti
                   ksqlVersionCheckerConfig,
                   serverRuntime,
                   moduleType,
-                  enableSettlingTime
+                  enableSettlingTime,
+                  activenessRegistrar
```

Comment:
Given that you pass a `Supplier<Boolean> hasActiveQueries` to this class, and you methods such as:

```
private boolean hasRecentRequests() {
    return (System.currentTimeMillis() - lastRequestTime) < MAX_LAST_REQUEST_AGE;
}

private boolean isActive() {
   return hasRecentRequests() || hasActiveQueries.get();
}
```

Then you can replace this with:
```suggestion
                  this::hasRecentRequests
```